internal/components/templates.templ

```diff
@@ -25,88 +26,97 @@ templ Build(goal, aim, username string) {
-				<input type="text" id={ fmt.Sprintf("%s-%s-%s", models.TextQuestion, id, order) } name={ fmt.Sprintf("%s-%s-%s", models.TextQuestion, id, order) } placeholder="Text input"/>
+templ BuildTemplateForm(username string, template views.TemplateView) {
+	@form(templ.Attributes{"hx-post": "/templates/save", "hx-swap": "outerHTML"}) {
+		<div class="form-group">
+			<label hidden for="title"></label>
+			@input(templ.Attributes{"type": "text", "id": "title", "name": "title", "value": template.Title, "placeholder": "Title of template"})
+			if template.TitleError != "" {
+				<span>{ template.TitleError }</span>
+			}
+		</div>
+		<div class="form-group">
+			<label hidden for="description"></label>
+			@textArea(templ.Attributes{"id": "description", "name": "description", "placeholder": "Description"}) {
+				{ template.Description }
+			}
+			if template.DescriptionError != "" {
+				<span>{ template.DescriptionError }</span>
+			}
+		</div>
+		<input type="hidden" name="username" value={ username }/>
+		<div data-sortable-right class="right-section">
+			for _, questionView := range template.QuestionViews {
+				<div class="draggable-input" data-type={ string(questionView.Question.Type) }>
+					@TemplateBuilderInput(questionView)
+				</div>
+			}
+		</div>
+	}
+}
+
+templ TemplateBuilderInput(questionView views.QuestionView) {
+	<div class="form-group">
+		switch questionView.Question.Type {
+			case models.TextQuestion:
+				<label hidden for={ fmt.Sprintf("%s-%s-%s", models.TextQuestion, questionView.Question.ID.Hex(), questionView.Question.OrderToString()) }></label>
+				@input(templ.Attributes{"type": "text", "id": fmt.Sprintf("%s-%s-%s", models.TextQuestion, questionView.Question.ID.Hex(), questionView.Question.OrderToString()), "name": fmt.Sprintf("%s-%s-%s", models.TextQuestion, questionView.Question.ID.Hex(), questionView.Question.OrderToString()), "value": questionView.Value, "placeholder": "Text input"})
 				<input disabled="disabled" type="text"/>
 				<button type="button" data-delete-row="true">X</button>
-			</div>
-		case models.NumberQuestion:
-			<div class="form-group">
-				<label hidden for={ fmt.Sprintf("%s-%s-%s", models.NumberQuestion, id, order) }></label>
-				<input type="text" id={ fmt.Sprintf("%s-%s-%s", models.NumberQuestion, id, order) } name={ fmt.Sprintf("%s-%s-%s", models.NumberQuestion, id, order) } placeholder="Number input"/>
+			case models.NumberQuestion:
+				<label hidden for={ fmt.Sprintf("%s-%s-%s", models.NumberQuestion, questionView.Question.ID.Hex(), questionView.Question.OrderToString()) }></label>
+				@input(templ.Attributes{"type": "text", "id": fmt.Sprintf("%s-%s-%s", models.NumberQuestion, questionView.Question.ID.Hex(), questionView.Question.OrderToString()), "name": fmt.Sprintf("%s-%s-%s", models.NumberQuestion, questionView.Question.ID.Hex(), questionView.Question.OrderToString()), "value": questionView.Value, "placeholder": "Number input"})
```

Comment:
since these are all text inputs we can probably put them in their own template and pass in the Number/Select etc type